From bd40b59f17696bacfd2479dd2b92a0248553a56e Mon Sep 17 00:00:00 2001
From: Kirk Wang <kirk.wang@amlogic.com>
Date: Mon, 10 Jul 2023 00:32:59 -0700
Subject: [PATCH] yocto: Screen recoder fail [1/1]

PD#SWPL-131322

Problem:
Screen recoder fail

Solution:
can not alloc memory from ion-fb

Verify:
t7

Change-Id: Ib68c65bc044056e7c5a8a465a771a8694b8f9859
Signed-off-by: kirk.wang <kirk.wang@amlogic.com>
---
 arch/arm64/boot/dts/amlogic/t7_a311d2_an400_debian.dts       | 2 +-
 arch/arm64/boot/dts/amlogic/t7_a311d2_an400_linux.dts        | 5 +++--
 arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_debian.dts      | 2 +-
 arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_linux.dts       | 5 +++--
 .../t7c_a311d2_an400_linux_drm_hdmitx_only_debian.dts        | 2 +-
 5 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_debian.dts b/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_debian.dts
index e42f3c25..65d534a4 100644
--- a/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_debian.dts
+++ b/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_debian.dts
@@ -187,7 +187,7 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			/* 1920x1080x4x6+3840x2160x4x3 round up 4M align */
-			size = <0x0 0x0>;
+			size = <0x0 0xa000000>;
 			alignment = <0x0 0x400000>;
 			alloc-ranges = <0x0 0x0 0x0 0xe0000000>;
 		};
diff --git a/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_linux.dts b/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_linux.dts
index e1b5bb43..7e838118 100644
--- a/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_linux.dts
+++ b/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_linux.dts
@@ -186,7 +186,8 @@
 		ion_fb_reserved:linux,ion-fb {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0x0 0x0>;
+			/* 1920x1080x4x6+3840x2160x4x3 round up 4M align */
+			size = <0x0 0xa000000>;
 			alignment = <0x0 0x400000>;
 			alloc-ranges = <0x0 0x0 0x0 0xe0000000>;
 		};
@@ -202,7 +203,7 @@
 		dmaheap_cma_reserved:heap-gfx {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0x0 0x10000000>;
+			size = <0x0 0xa000000>;
 			alignment = <0x0 0x400000>;
 		};
 		dmaheap_fb_reserved:heap-fb {
diff --git a/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_debian.dts b/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_debian.dts
index 86901583..221f3598 100644
--- a/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_debian.dts
+++ b/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_debian.dts
@@ -177,7 +177,7 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			/* 1920x1080x4x6+3840x2160x4x3 round up 4M align */
-			size = <0x0 0x0>;
+			size = <0x0 0xa000000>;
 			alignment = <0x0 0x400000>;
 			alloc-ranges = <0x0 0x0 0x0 0xe0000000>;
 		};
diff --git a/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_linux.dts b/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_linux.dts
index 239f8287..4975ffe8 100644
--- a/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_linux.dts
+++ b/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_linux.dts
@@ -175,7 +175,8 @@
 		ion_fb_reserved:linux,ion-fb {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0x0 0x0>;
+			/* 1920x1080x4x6+3840x2160x4x3 round up 4M align */
+			size = <0x0 0xa000000>;
 			alignment = <0x0 0x400000>;
 			alloc-ranges = <0x0 0x0 0x0 0xe0000000>;
 		};
@@ -197,7 +198,7 @@
 		dmaheap_cma_reserved:heap-gfx {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0x0 0x10000000>;
+			size = <0x0 0xa000000>;
 			alignment = <0x0 0x400000>;
 		};
 		dmaheap_fb_reserved:heap-fb {
diff --git a/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_linux_drm_hdmitx_only_debian.dts b/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_linux_drm_hdmitx_only_debian.dts
index 82a1bdaf..f7d88a1b 100644
--- a/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_linux_drm_hdmitx_only_debian.dts
+++ b/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_linux_drm_hdmitx_only_debian.dts
@@ -178,7 +178,7 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			/* 1920x1080x4x6+3840x2160x4x3 round up 4M align */
-			size = <0x0 0x0>;
+			size = <0x0 0xa000000>;
 			alignment = <0x0 0x400000>;
 			alloc-ranges = <0x0 0x0 0x0 0xe0000000>;
 		};
-- 
2.25.1

